fraud-detection-generate-maps/main.py

```diff
@@ -0,0 +1,162 @@
+    return gee_mapbiomas_map
+
+def generate_full_map_and_upload_to_gcs(lat, lon, land_use_mapbiomas_image, mapbiomas_land_use_base_map, document_id):
+    """
+    Generates a map of a given location using the Google Earth Engine (GEE) Python API and uploads the map to Google Cloud Storage (GCS).
+    
+    Args:
+        lat (float): The latitude of the location in decimal degrees.
+        lon (float): The longitude of the location in decimal degrees.
+        land_use_mapbiomas_image (ee.Image): An `ee.Image` object representing the land use map.
+        mapbiomas_land_use_base_map (geemap.Map): A `geemap.Map` object representing the base map.
+        document_id (str): The ID of the Firestore document.
+    
+    Returns:
+        None
+    """
+    # create a 1 km and 10 km radius buffer around the point
+    point = ee.Geometry.Point(lon, lat)
+    radius_1km_buffer = point.buffer(1000)
+    radius_10km_buffer = point.buffer(10000)
+
+    if len(mapbiomas_land_use_base_map.layers) > 2:
+        mapbiomas_land_use_base_map.remove_ee_layer(RADIUS_MASK_1_KM_LAYER_NAME)
+        mapbiomas_land_use_base_map.remove_ee_layer(RADIUS_MASK_10_KM_LAYER_NAME)
+
+    # create a mask for the 10 km radius buffer with a 10km hole
+    mapbiomas_land_use_base_map.addLayer(land_use_mapbiomas_image.mask().geometry().difference(radius_10km_buffer), {}, RADIUS_MASK_10_KM_LAYER_NAME)
+    # add the 1 km radius buffer as a layer
+    mapbiomas_land_use_base_map.addLayer(radius_1km_buffer, {"color": "white"}, RADIUS_MASK_1_KM_LAYER_NAME)
+    # center the map on the point
+    mapbiomas_land_use_base_map.centerObject(radius_10km_buffer, 11)
+    
+    # generate the html file and upload to GCS
+    filename = generate_html_file_from_map(mapbiomas_land_use_base_map, document_id)
+    upload_map_to_gcs(filename, document_id)
+
+def generate_html_file_from_map(gee_mapbiomas_map, document_id):
+    """
+    Generates an HTML file from a Google Earth Engine (GEE) map and saves it to disk.
+    
+    Args:
+        gee_map (geemap.Map): A `geemap.Map` object representing the map to convert to HTML.
+        document_id (str): The ID of the Firestore document.
+    
+    Returns:
+        The full filename of the generated HTML file.
+    """
+
+    download_dir = os.getcwd()
+    html_file = os.path.join(download_dir, f"{document_id}.html")
+    gee_mapbiomas_map.to_html(
+        filename=html_file, title="Land Use details", width="100%", height="300px"
+    )
+    return html_file
+
+def upload_map_to_gcs(html_file, document_id):
+    """
+    Uploads an HTML file to Google Cloud Storage (GCS).
+    
+    Args:
+        html_file (str): The full filename of the HTML file to upload.
+        document_id (str): The ID of the Firestore document.
+    
+    Returns:
+        None
+    """
+    storage_client = storage.Client()
+    bucket = storage_client.bucket("timberid-maps")
+    blob = bucket.blob(document_id)
```

Comment:
are you using a naming convention to locate the file on the frontend side?
If so, we should have a naming convention that appends the particular analysis being made so that this can be cloned and done for a different analysis.
eg
    bucket = storage_client.bucket("timberid-maps/" + document_id )
    blob = bucket.blob("land_use_map.html")